nova/virt/vmwareapi/volumeops.py

```diff
@@ -419,6 +419,53 @@ def _attach_volume_iscsi(self, connection_info, instance,
                                profile_id=data.get('profile_id'))
         LOG.debug("Attached ISCSI: %s", connection_info, instance=instance)
 
+    def _get_controller_key_and_unit(self, vm_ref, adapter_type):
+        LOG.debug("_get_controller_key_and_unit vm: %(vm_ref)s, adapter: "
+                  "%(adapter)s.",
+                  {'vm_ref': vm_ref, 'adapter': adapter_type})
+        client_factory = self._session.vim.client.factory
+        devices = self._session._call_method(vutil,
+                                             "get_object_property",
+                                             vm_ref,
+                                             "config.hardware.device")
```

Comment:
we have `devices = vm_util.get_hardware_devices(self._session, vm_ref)` for that